copy.go

```diff
@@ -342,5 +342,23 @@ func prepareCopy(ctx context.Context, dst Target, dstRef string, proxy *cas.Prox
 		}
 	}
 
+	onCopySkipped := opts.OnCopySkipped
+	opts.OnCopySkipped = func(ctx context.Context, desc ocispec.Descriptor) error {
+		if onCopySkipped != nil {
+			if err := onCopySkipped(ctx, desc); err != nil {
+				return err
+			}
+		}
+		if !descriptor.EqualOCI(desc, root) {
+			return nil
+		}
+		// copy reference for skipped root
+		if refPusher, ok := dst.(registry.ReferencePusher); ok {
+			return copyCachedNodeWithReference(ctx, proxy, refPusher, desc, dstRef)
+		} else {
+			return dst.Tag(ctx, root, dstRef)
+		}
```

Comment:
```suggestion
		}
		return dst.Tag(ctx, root, dstRef)

```

We can save one more line.